schedule/yast/autoyast/autoyast_btrfs_quota.yaml

```diff
@@ -46,14 +46,13 @@ test_data:
           unique_key: device
           device: /dev/vda
           partitions:
-            - partition:
-                unique_key: mount
-                mount: swap
+            - count_off
```

Comment:
I'm not sure if it's the best approach we can find. We already have special keys, like `unique_key`, why not to have this option there, at least, which will be applied to all children nodes.
Here we have a list and `count_off` is not a partition, so we are mixing data which serves different to purpose.

Reply:
unique key is related to one item of the list and this flag is for the list itself, I understand your point, I considered, but didn't find better solution at the end.

Reply:
Yes, I thinks this approach doesn't help much and in couple of months we will forget about this syntax.
We can also rethink whole structure, as xml has attributes for nodes, I'm not even sure if current code support checks any attribute validation.

I has something like this in mind, that we can differentiate between node/attribute values and some properties we define, like you already did for the search key (I still not sure if we need it).
```
- partitions:
  unique_key: mount
  count:2
  items:
  - name: /dev/sda1
```
Also, tbh instead of some hacky value in the array I would rather force defining all expected nodes and be strict with it.

Reply:
Let's discuss it after the daily tomorrow. I'm sure we can find better way.

Reply:
I would like to also tell my opinion on this matter:)

I think that counting should be explicit, i.e. if I want to count the number of partitions, then I provide the amount with test_data, otherwise, don't count.

We even can extract counting into separate method and call it from test module in case we want to count the amount of partitions. This will also may allow to make the methods more transparent (I already stated this in the ticket as "optional" task).

So, I would prefer Rodion's solution or something similar to that. I would avoid using some special keys in yaml as much as possible.

Reply:
For inspiration, we can take a look on some existing xml to yaml converters https://beautifytools.com/xml-to-yaml-converter.php 
There they use following structure to express attributes/value difference:
```yaml
  do_registration: 
   "_config:type": boolean
   __text: true
```